src/main/java/net/teamterminus/machineessentials/energy/electric/base/ElectricWireBlockEntity.java

```diff
@@ -0,0 +1,72 @@
+package net.teamterminus.machineessentials.energy.electric.base;
+
+import lombok.Getter;
+import net.minecraft.block.entity.BlockEntity;
+import net.modificationstation.stationapi.api.util.math.Direction;
+import net.modificationstation.stationapi.api.util.math.Vec3i;
+import net.teamterminus.machineessentials.MachineEssentials;
+import net.teamterminus.machineessentials.energy.electric.api.Electric;
+import net.teamterminus.machineessentials.energy.electric.api.ElectricWire;
+import net.teamterminus.machineessentials.energy.electric.api.WireProperties;
+import net.teamterminus.machineessentials.network.Network;
+import net.teamterminus.machineessentials.network.NetworkType;
+import net.teamterminus.machineessentials.util.AveragingCounter;
+import net.teamterminus.machineessentials.util.BlockEntityInit;
+
+public abstract class ElectricWireBlockEntity extends BlockEntity implements BlockEntityInit, ElectricWire {
+
+    public Network energyNet;
+    @Getter
+    protected WireProperties properties;
+    protected long voltageRating = 0;
+    protected long ampRating = 0;
+
+    protected AveragingCounter averageAmpLoad = new AveragingCounter();
+    @Getter
+    protected long temperature = 0;
```

Comment:
This isn used, might be nice to atleast leave it public/have a setter so mods can implement a functionality